src/xtgeo/grid3d/_gridprop_value_init.py

```diff
@@ -1,101 +1,141 @@
 """GridProperty (not GridProperies) some etc functions"""
+from __future__ import annotations
 
-# from ._gridprop_import_eclrun import import_eclbinary
-#  --> INIT, UNRST
-#
-# from ._gridprop_import_grdecl import import_grdecl_prop, import_bgrdecl_prop
-#  --> ASCII and BINARY GRDECL format
-# from ._gridprop_import_roff import import_roff
-#  --> BINARY ROFF format
-
+from typing import TYPE_CHECKING, Optional, Tuple, Union
 
 import numpy as np
 
 import xtgeo
+from xtgeo.common import XTGeoDialog
 
-xtg = xtgeo.common.XTGeoDialog()
+logger = XTGeoDialog().functionlogger(__name__)
 
-logger = xtg.functionlogger(__name__)
+if TYPE_CHECKING:
+    from xtgeo.grid3d import Grid, GridProperty
 
 
-def gridproperty_non_dummy_values(gridlike, dimensions, values, isdiscrete):
+def gridproperty_non_dummy_values(
+    gridlike: Optional[Union[Grid, GridProperty]],
+    dimensions: Tuple[int, int, int],
+    values: Optional[Union[np.ndarray, float, int]],
+    isdiscrete: bool,
+) -> np.ma.MaskedArray:
     """Gives the initial values array of an gridprop.
 
-    Param:
-      gridlike: Either Grid or GridProperty, giving the mask to replicate.
-      dimensions: The dimensions of the gridprop
-      values: The values parameter given to init.
-      isdiscrete: The discrete parameter given to init.
+    Parameters:
+        gridlike:
+            Either Grid or GridProperty, giving the mask to replicate.
+        dimensions:
+            The (ncol, nrow, nlay) dimensions of the grid property.
+        values:
+            The values parameter given to init.
+        isdiscrete:
+            The discrete parameter given to init.
```

Comment:
Proposing this format for docstrings going forward. 

- The type no longer needs to be given here, it is inferred from the method/function signature. 
- Various things can be used instead of `Parameters` -- `Args` is frequent in the code, which is an internal alias for `Parameters`. Parameters imo is the pedantically correct term.
- The new line between the param name and description is not needed, but my opinion is that it's more readable.

Reply:
Looking at just in the editor it like it. I'm not to use to these kinds of doc. strings, so i do not know this propagates to sphinx.